### PR TITLE
Refactor series-area handling out of Chart.

### PR DESCRIPTION
**Pull request by alantreadway**
Move all series-area related interaction handling out into a new `SeriesAreaManager`.

The intent in this change is just to lift-and-shift, not rework (or at least to minimise this).

### CODE REVIEW

File: packages/ag-charts-community/src/chart/flowProportionChart.ts [hunk 1]
```diff
@@ -12,6 +12,10 @@ export class FlowProportionChart extends Chart {
     static readonly className = 'FlowProportionChart';
     static readonly type = 'flow-proportion' as const;
 
+    override getChartType() {
+        return 'flow-proportion' as const;
```
Comment (jacobp100) [inline packages/ag-charts-community/src/chart/flowProportionChart.ts:16]: Could this not on the chart base-class as `return this.constructor.type`?

File: packages/ag-charts-community/src/chart/series/seriesAreaManager.ts [hunk 2]
```diff
@@ -0,0 +1,630 @@
...
+    }
+
+    private handleSeriesFocus(otherIndexDelta: number, datumIndexDelta: number) {
+        if (this.chartType === 'hierarchy') {
+            this.handleHierarchySeriesFocus(otherIndexDelta, datumIndexDelta);
+            return;
+        }
+        const { focus, seriesRect, series } = this;
+        const visibleSeries = series.filter((s) => s.visible);
+        if (visibleSeries.length === 0) return;
```
Comment (jacobp100) [inline packages/ag-charts-community/src/chart/series/seriesAreaManager.ts:534]:
```suggestion
        const hasVisibleSeries = series.some((s) => s.visible);
        if (!hasVisibleSeries) return;
```

**Review by jacobp100 — APPROVED** [review]

File: packages/ag-charts-community/src/chart/flowProportionChart.ts (hunk 1 above)
Comment (alantreadway) [reply packages/ag-charts-community/src/chart/flowProportionChart.ts:16]: Yes, but let's do it in a follow-up as this is a big change.

**Comment by alantreadway** [comment]
Superceeded by https://github.com/ag-grid/ag-charts/pull/2025

**Pull request closed without merging**